### DISCUSSION TITLE
ValueError: operands could not be broadcast together with shapes (5553,) (5551,)

### DISCUSSION THREAD

**Discussion by brentmsmith** [mboquien/cigale#33] [Q&A] [answered]
Hello, I keep getting the above error and I'm not sure why. I have other pcigale.ini files that give me no issue. Below is the full trace:

───────────────────── Traceback (most recent call last) ──────────────────────╮
│ /home/user/.local/lib/python3.6/site-packages/pcigale:11 in <module>     │
│                                                                              │
│    8 if __name__ == '__main__':                                              │
│    9 │   sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])   │
│   10 │   sys.exit(                                                           │
│ ❱ 11 │   │   load_entry_point('pcigale', 'console_scripts', 'pcigale')()     │
│   12 │   )                                                                   │
│   13                                                                         │
│                                                                              │
│ /home/user/cigale/pcigale/__init__.py:139 in main                        │
│                                                                              │
│   136 │   │   elif args.parser == "check":                                   │
│   137 │   │   │   check(config)                                              │
│   138 │   │   elif args.parser == "run":                                     │
│ ❱ 139 │   │   │   run(config)                                                │
│   140                                                                        │
│                                                                              │
│ /home/user/cigale/pcigale/__init__.py:71 in run                          │
│                                                                              │
│    68 │   │   console.print(f"{INFO} Start: {start.isoformat('/', 'seconds') │
│    69 │   │   start = time.monotonic()  # Simpler time for run duration      │
│    70 │   │                                                                  │
│ ❱  71 │   │   analysis_module.process(configuration)                         │
│    72 │   │                                                                  │
│    73 │   │   end = dt.datetime.now()                                        │
│    74 │   │   console.print(f"{INFO} End: {end.isoformat('/', 'seconds')}")  │
│                                                                              │
│ /home/user/cigale/pcigale/analysis_modules/pdf_analysis/__init__.py:235  │
│ in process                                                                   │
│                                                                              │
│   232 │   │   obs = ObservationsManager(conf, params)                        │
│   233 │   │   obs.save("observations")                                       │
│   234 │   │                                                                  │
│ ❱ 235 │   │   results = self._compute(conf, obs, params)                     │
│   236 │   │   console.print(f"{INFO} Sanity check of the analysis results.") │
│   237 │   │   results.best.analyse_chi2()                                    │
│   238                                                                        │
│                                                                 

             │
│ /home/user/cigale/pcigale/analysis_modules/pdf_analysis/__init__.py:182  │
│ in _compute                                                                  │
│                                                                              │
│   179 │   │   │   # avoid recomputing the models when we do a mock analysis  │
│   180 │   │   │   if not hasattr(self, "_models"):                           │
│   181 │   │   │   │   console.print(f"{INFO} Computing models.")             │
│ ❱ 182 │   │   │   │   models = self._compute_models(conf, obs, params, ibloc │
│   183 │   │   │   │   if nblocks == 1:                                       │
│   184 │   │   │   │   │   self._models = models                              │
│   185 │   │   │   else:                                                      │
│                                                                              │
│                                                                              │
│ /home/user/cigale/pcigale/analysis_modules/pdf_analysis/__init__.py:113  │
│ in _compute_models                                                           │
│                                                                              │
│   110 │   │   │   params.blocks[iblock],                                     │
│   111 │   │   │   initargs,                                                  │
│   112 │   │   │   init_worker_sed,                                           │
│ ❱ 113 │   │   │   conf["cores"]                                              │
│   114 │   │   )                                                              │
│   115 │   │                                                                  │
│   116 │   │   # Print the final value as it may not otherwise be printed     │
│                                                                              │
│ /home/user/cigale/pcigale/analysis_modules/pdf_analysis/__init__.py:168  │
│ in _parallel_job                                                             │
│                                                                              │
│   165 │   │   │   with mp.Pool(                                              │
│   166 │   │   │   │   processes=ncores, initializer=initializer, initargs=in │
│   167 │   │   │   ) as pool:                                                 │
│ ❱ 168 │   │   │   │   pool.starmap(worker, enumerate(items), chunksize)      │
│   169 │   │   │                                                              │
│   170 │   │   │   # After the parallel processes have exited, it can be rest │
│   171 │   │   │   counter.progress = progress                                │
│                                                                              │
│ /usr/lib64/python3.6/multiprocessing/pool.py:274 in starmap                  │
│                                                                              │
│   271 │   │   be iterables as well and will be unpacked as arguments. Hence  │
│   272 │   │   `func` and (a, b) becomes func(a, b).                          │
│   273 │   │   '''                                                            │
│ ❱ 274 │   │   return self._map_async(func, iterable, starmapstar, chunksize) │
│   275 │                                                                      │
│   276 │   def starmap_async(self, func, iterable, chunksize=None, callback=N │
│   277 │   │   │   error_callback=None):                                      │
│                                                                              │
│ /usr/lib64/python3.6/multiprocessing/pool.py:644 in get                      │
│                                                                              │
│   641 │   │   if self._success:                                              │
│   642 │   │   │   return self._value                                         │
│   643 │   │   else:                                                          │
│ ❱ 644 │   │   │   raise self._value                                          │
│   645 │                                                                      │
│   646 │   def _set(self, i, obj):                                            │
│   647 │   │   self._success, self._value = obj                               │
╰──────────────────────────────────────────────────────────────────────────────╯
ValueError: operands could not be broadcast together with shapes (5553,) (5551,)
[pcigale.ini.spec](https://github.com/mboquien/cigale/files/9992175/pcigale.ini.spec.txt)
[pcigale.ini](https://github.com/mboquien/cigale/files/9992176/pcigale.ini.txt)

**Comment by mboquien** [comment, marked as answer]
The order of the modules is incorrect. The `redshifting` module must come last.

**Reply by tranthithai** [reply]
I don't think so. Last time, when I used Cigale to estimate other parameters (fixing spectroscopic redshift), the cigale ran properly. But now, I want to estimate photometric redshift, I  set a negative value of redshift in the input file and set a grid for the [redshiffting] . I'm getting the same problem as mentioned. Do you have any advice ?

**Reply by mboquien** [reply]
If you have an issue, please open a new ticket.
